daemon/kill.go

```diff
@@ -179,7 +179,9 @@ func (daemon *Daemon) Kill(container *containerpkg.Container) error {
 // killPossibleDeadProcess is a wrapper around killSig() suppressing "no such process" error.
 func (daemon *Daemon) killPossiblyDeadProcess(container *containerpkg.Container, sig int) error {
 	err := daemon.killWithSignal(container, sig)
```

Comment:
Actually killWithSignal hangs if the container has "IsRunning()" returning true while having pid = 0. We should not fall into this state anymore on startup with the fix, but I think it would be nice to make killWithSignal more robust ?

Reply:
Yes, it is a terrible mess.